File: core/src/ru/myitschool/cubegame/math/StatementAction.java
package ru.myitschool.cubegame.math;

/**
 * Created by Voyager on 18.04.2018.
 */
public class StatementAction extends MathAction {

    private Statement statement;
    private MathAction trueAction;
    private MathAction falseAction;

    public StatementAction(Statement statement, MathAction trueAction, MathAction falseAction) {
        this.statement = statement;
        this.trueAction = trueAction;
        this.falseAction = falseAction;
    }

    @Override
    public int act() {
        if (statement.result()){
            return trueAction.act();
        } else {
            return falseAction.act();
        }
    }

    @Override
    public String getDescription() {
        return "if(statement(" + statement.getDescription() + "), true(" + trueAction.getDescription() + "), false(" + falseAction.getDescription() + ")";
    }
}

File: core/src/com/wolg_vlad/dcrawler/math/StatementAction.kt
package com.wolg_vlad.dcrawler.math

/**
 * Created by Voyager on 18.04.2018.
 */
class StatementAction(private val statement: Statement, private val trueAction: MathAction, private val falseAction: MathAction) : MathAction() {
    override fun act(): Int {
        return if (statement.result()) {
            trueAction.act()
        } else {
            falseAction.act()
        }
    }

    override val description: String?
        get() = "if(statement(" + statement.description + "), true(" + trueAction.description + "), false(" + falseAction.description + ")"

}